Fix :Man on OpenBSD

jbernard opened the pull request:
OpenBSD's man returns all candidates with -w instead of the first one.
This patch takes the first candidate in the case where multiple are
found.

Fixes: #8341

This is the first non-personal vimscript I've ever written, so I suspect there's a much better way to do this.  But it does solve the issue so hopefully it can inspire someone more knowledgeable.

nhooyr reviewed (changes requested):
Nice!

Here are some improvements.

Instead of trimming the path arg elsewhere, do it before its returned in https://github.com/neovim/neovim/blob/ebb1acb3c083e63667c3606233a496b11d3faa83/runtime/autoload/man.vim#L214

And also add a comment with the issue number explaining why we're trimming.

justinmk commented on runtime/autoload/man.vim:
```diff
@@ -213,7 +213,11 @@ endfunction
+    " from the search command, so we split here to make sure we only
+    " take the first one.  Discovered and reported in
+    " https://github.com/neovim/neovim/issues/8341
+    return split(s:system(['man', s:find_arg, a:name]))[0]
```
I guess split() could be empty, so should use `get()` instead of `[0]`:

      return get(split(s:system(['man', s:find_arg, a:name])), 0, '')

Also as @nhooyr mentioned the newline trim should be done here. So:

    return substitute(get(split(s:system(['man', s:find_arg, a:name])), 0, ''), '\n\+$', '', '')

nhooyr replied:
You'll need to remove the slice here that takes out the newline: https://github.com/jbernard/neovim/blob/a5dcfa1b7e07069ca5bcd0e9840adaa0e4138353/runtime/autoload/man.vim#L22

you'll also need to trim the newline in the second branch of `s:get_path`: https://github.com/jbernard/neovim/blob/a5dcfa1b7e07069ca5bcd0e9840adaa0e4138353/runtime/autoload/man.vim#L227

nhooyr commented on runtime/autoload/man.vim:
```diff
@@ -233,13 +237,16 @@ function! s:verify_exists(sect, name) abort
   " ':Man PRIntf', we still want the name of the buffer to be 'printf' or
   " whatever the correct capitilization is.
-  let path = path[:len(path)-2]
+  let path = path[:len(path) - 1]
```
This needs to be removed. It was supposed to trim the trailing newline but that should be done in `s:get_path`.

jbernard committed:
Fix :Man on OpenBSD

OpenBSD's man returns all candidates when searching with -w instead of
the first one it finds.  So this patch takes the first one if multiple
entries are found to restore the expected behaviour.

Fixes: #8341

justinmk commented:
merged. there was another `s:get_path` call site, I fixed it as well.

nhooyr commented:
There is a bug in f1a3075
